app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,38 @@
 class PostsController < ApplicationController
+  before_action :set_user, only: %i[index new create]
+
   def index
-    @user = User.find(params[:user_id])
-    @posts = Post.where(author_id: params[:user_id]).order(id: :asc)
-    @posts = @posts.paginate(page: params[:page])
+    @posts = @user.posts.build
+    @posts = @user.posts.paginate(page: params[:page])
   end
 
   def show
     @post = Post.find(params[:id])
   end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    Rails.logger.debug "POST PARAMS: #{post_params.inspect}"
+    @post = @user.posts.build(post_params)
```

Comment:
- [ ] Hey there! 🌟 Awesome job on implementing the create action for post creation! 🚀 As specified in the project requirements, let's make sure to utilize the current_user you've set up in the ApplicationController for creating the post. This adjustment is crucial to align with the project specifications. 🔄 Keep up the great work! 💻👏

```suggestion
    @post = @current_user.posts.new(post_params)
```